### PR TITLE
Merge CodeAction.CustomTags into CodeAction.Tags

### PR DESCRIPTION
**Pull request by JoeRobich**
1. Removes CustomTags and provides way to add to CodeAction.Tags 
2. Updates CodeFixService and CodeRefactoringService to add a prefix to the provider name when adding a Tag
3. Updates places where CodeAction.Tags is overriden to Concat the `base.Tags` collection
4. Updates the Razor predefined names with the expect prefix

CC: @jasonmalinowski , @TanayParikh

### CODE REVIEW

**Comment by TanayParikh** [comment]
cc/ @NTaylorMullen this'll require a reaction on the Razor Tooling end (CustomTags -> Tags)

**Comment by NTaylorMullen** [comment]
@JoeRobich we'll need to do a joint insertion then. Let me know when an insertion PR with these bits is available and i'll roll a Razor one to include in it 😄

File: src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs [hunk 1]
```diff
@@ -180,10 +182,10 @@ private ConcatImmutableArray<CodeRefactoringProvider> GetProviders(Document docu
                         // Serialize access for thread safety - we don't know what thread the refactoring provider will call this delegate from.
                         lock (actions)
                         {
-                            // Add the Refactoring Provider Name to the parent CodeAction's CustomTags.
+                            // Add the Refactoring Provider Name to the parent CodeAction's Tags.
                             // Always add a name even in cases of 3rd party refactorings that do not export
                             // name metadata.
-                            action.AddCustomTag(providerMetadata?.Name ?? provider.GetTypeDisplayName());
+                            action.AddTag($"{ProviderTagPrefix}{providerMetadata?.Name ?? provider.GetTypeDisplayName()}");
```
Comment (TanayParikh) [inline src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs:188]: Note with these prefixes, Razor may no longer need to use reflection to distinguish code fix names from other tags. cc/ @NTaylorMullen

**Review by TanayParikh — COMMENTED** [review]
LGTM 👍

**Pull request closed without merging**